app/code/Magento/Elasticsearch/Elasticsearch5/SearchAdapter/Query/Builder.php

```diff
@@ -83,18 +85,18 @@ public function initQuery(RequestInterface $request)
     {
         $dimension = current($request->getDimensions());
         $storeId = $this->scopeResolver->getScope($dimension->getValue())->getId();
-
         $searchQuery = [
             'index' => $this->searchIndexNameResolver->getIndexName($storeId, $request->getIndex()),
             'type' => $this->clientConfig->getEntityType(),
             'body' => [
-                'from' => $request->getFrom(),
+                'from' => $this->getFrom($request),
```

Comment:
I think inline looks a bit cleaner. What do you thing?
```suggestion
                'from' => min((int)$request->getFrom(), self::ELASTIC_MAX_INT),
```

PS What about negative values? Should we also handle it?

Reply:
Agree. Will update. But don't think that we need to handle negative values. In that case, elastic returns exception with message numHits must be > 0